tbdex/rfq/rfq.go

```diff
@@ -166,6 +172,213 @@ func (r *RFQ) verifyPrivateData() error {
 	return nil
 }
 
+// VerifyOfferingRequirements verifies that the RFQ meets all the requirements of the offering,
+// including payin/payout amount, method, and claims.
+func (r *RFQ) VerifyOfferingRequirements(offering offering.Offering) error {
+	if offering.Metadata.Protocol != r.Metadata.Protocol {
+		return fmt.Errorf("protocol version mismatch. (rfq) %s != %s (offering)", r.Metadata.Protocol, offering.Metadata.Protocol)
+	}
+
+	if offering.Metadata.ID != r.Data.OfferingID {
+		return fmt.Errorf("offering ID mismatch. (rfq) %s != %s (offering)", r.Data.OfferingID, offering.Metadata.ID)
+	}
+
+	// Verifying payin amount is less than maximum
+	payinAmount, err := strconv.ParseFloat(r.Data.Payin.Amount, 64)
+	if err != nil {
+		return fmt.Errorf("failed to parse payin amount: %w", err)
+	}
+
+	if offering.Data.Payin.Max != "" {
+		maxAmount, err := strconv.ParseFloat(offering.Data.Payin.Max, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse offering payin max amount: %w", err)
+		}
+
+		if payinAmount > maxAmount {
+			return fmt.Errorf("payin amount exceeds maximum. (rfq) %f > %f (offering)", payinAmount, maxAmount)
+		}
+	}
+
+	// Verify payin amount is greater than minimum
+	if offering.Data.Payin.Min != "" {
+		minAmount, err := strconv.ParseFloat(offering.Data.Payin.Min, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse offering payin min amount: %w", err)
+		}
```

Comment:
we should use [this decimal lib](https://pkg.go.dev/github.com/shopspring/decimal) instead of `strconv.ParseFloat`